### ISSUE TITLE
[1.12.2] CraftTweaker Script conflict still occurs, using gunpowder for any potion converts them to Splash Potion of Virus Protection.

### ISSUE THREAD

**Issue by SonicX8000** [TheUnderTaker11/GeneticsReborn#182]
Minecraft: 1.12.2
Forge: 14.23.5.2847

Mods: 8
BrandonsCore-1.12.2-2.4.18.210-universal
CodeChickenLib-1.12.2-3.2.3.358-universal
CraftTweaker2-1.12-4.1.20.562
Draconic-Evolution-1.12.2-2.3.25.351-universal
geneticsreborn-1.12-1.27
jei_1.12.2-4.15.0.291
RedstoneFlux-1.12-2.1.1.1-universal
theoneprobe-1.12-1.4.28

=====

The CT script being used:
https://drive.google.com/file/d/1PXeyFnGxc8EDzjl39pSFD1QX9Vj0nN0h/view?usp=sharing

Even with the Potion IDs being changed the conflict still occurs where using gunpowder for any potion will always result in Splash Potion of Virus Protection.

One interesting find is that I had this crafttweaker script added but forgot to include the Potion Core mod when testing so when I used Water Bottles & Gunpowder in a brewing stand... it brew'd a Splash Potion of Virus Protection so it's not actually a conflict with Potion Core.

Looking in the source code, for "GRPotion.java"... there's this section of code here...

-----
		BrewingRecipeRegistry.addRecipe(PotionUtils.addPotionToItemStack(new ItemStack(Items.POTIONITEM), CURE_POTION), 
				new ItemStack(Items.GUNPOWDER, 1),
				PotionUtils.addPotionToItemStack(new ItemStack(Items.SPLASH_POTION), CURE_POTION));
-----

By removing this line of code... the conflict with the crafttweaker script no longer occurs, meaning that potions will brew into their respectful Splash Potions of 'potion_name_here' instead of Splash Potion of Virus Protection as well as converting Virus Protection Potions into their Splash Potion versions without issue if using gun powder.

I am not sure if that line of code is used for anything else as I think it only adds in the recipe.

*EDIT*
I tried removing any recipes or sections that requires gunPowder in the crafttweaker script and the conflict still occurs. Potion Core was not added either so somehow it still conflicts despite no gunpowder being listed.

**Comment by clubpetey** [comment]
While removing the line of code you mention fixes the CT integration, it breaks the functions of those potions for those that do not have CT.  Forge/MC has code that auto-creates Splash, Lingering, Arrow versions of all potions in the game.  I cannot stop this and this is why you see a cure arrow in creative, these arrows/potions will not work however, because they use the standard potion logic and the GR potions are not implemented with standard logic.

https://drive.google.com/file/d/1AMv-LlUoxydOgdkpRB_ryL0m7QQ7kwMC/view?usp=sharing

I've custom built a version of the mod above that switches the base class of the GR potions to something other than the default MC potion, in hopes that it fixes your issue.  As the gunpowder recipe only applies to my potion class now. Please replace the 1.27 you have on your system with this one and let me know if it fixes the gunpowder problem.

Also, this patch fixes the ability to brew cure potions, as the Emerald Heart helix works correctly now.  

If this fixes your issues, I will release this version as 1.28

**Comment by SonicX8000** [comment]
Sorry for the wait, just woke up.

I've tested the custom version that you provided in a new world as well as in my private survival world and I can confirm that the conflict no longer occurs. Potions brew into their correct splash versions when using gun powder.

Also... ah, I was going to mention something with the Emerald Heart Helix but forgot to do that. Can also confirm that the recipe works now as I was able to brew Virus Protection Potions
